Improve error message when multiple hibernate interceptors are found

### Description

Can this message please be expanded to include the fqdn of the interceptors

```
__  ____  __  _____   ___  __ ____  ______                                                                                                                                                                                                                                           
 --/ __ \/ / / / _ | / _ \/ //_/ / / / __/
 -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \
--\___\_\____/_/ |_/_/|_/_/|_|\____/___/
2024-07-19 08:41:20,146 WARN  [org.hib.dia.Dialect] (Quarkus Main Thread) HHH000511: The 11.5.0 version for [org.hibernate.dialect.PostgreSQLDialect] is no longer supported, hence certain features may not work properly. The minimum supported version is 12.0.0. Check the community dialects project for available legacy versions.

2024-07-19 08:41:21,902 ERROR [io.qua.run.Application] (Quarkus Main Thread) Failed to start application (with profile [dev]): java.lang.RuntimeException: Failed to start quarkus
        at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
        at io.quarkus.runtime.Application.start(Application.java:101)
        at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:111)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:71)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:44)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:124)
        at io.quarkus.runner.GeneratedMain.main(Unknown Source)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at io.quarkus.runner.bootstrap.StartupActionImpl$1.run(StartupActionImpl.java:113)
        at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: jakarta.persistence.PersistenceException: [PersistenceUnit: <default>] Unable to build Hibernate SessionFactory
        at io.quarkus.hibernate.orm.runtime.boot.FastBootEntityManagerFactoryBuilder.persistenceException(FastBootEntityManagerFactoryBuilder.java:129)
        at io.quarkus.hibernate.orm.runtime.boot.FastBootEntityManagerFactoryBuilder.build(FastBootEntityManagerFactoryBuilder.java:89)
        at io.quarkus.hibernate.orm.runtime.FastBootHibernatePersistenceProvider.createEntityManagerFactory(FastBootHibernatePersistenceProvider.java:72)
        at jakarta.persistence.Persistence.createEntityManagerFactory(Persistence.java:80)
        at jakarta.persistence.Persistence.createEntityManagerFactory(Persistence.java:55)
        at io.quarkus.hibernate.orm.runtime.JPAConfig$LazyPersistenceUnit.get(JPAConfig.java:154)
        at io.quarkus.hibernate.orm.runtime.JPAConfig$1.run(JPAConfig.java:61)
        ... 1 more
Caused by: java.lang.IllegalStateException: Multiple instances of Interceptor were found for persistence unit <default>. At most one instance can be assigned to each persistence unit.
        at io.quarkus.hibernate.orm.runtime.PersistenceUnitUtil.singleExtensionInstanceForPersistenceUnit(PersistenceUnitUtil.java:33)
        at io.quarkus.hibernate.orm.runtime.boot.FastBootEntityManagerFactoryBuilder.populate(FastBootEntityManagerFactoryBuilder.java:206)
        at io.quarkus.hibernate.orm.runtime.boot.FastBootEntityManagerFactoryBuilder.build(FastBootEntityManagerFactoryBuilder.java:85)
        ... 6 more

```


### Implementation ideas

_No response_

--- a/extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/boot/AmbiguousPersistenceUnitExtensionTest.java
+++ b/extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/boot/AmbiguousPersistenceUnitExtensionTest.java
@@ -1,0 +1,43 @@
+package io.quarkus.hibernate.orm.boot;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.hibernate.Interceptor;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.hibernate.orm.MyEntity;
+import io.quarkus.hibernate.orm.PersistenceUnitExtension;
+import io.quarkus.test.QuarkusUnitTest;
+
+public class AmbiguousPersistenceUnitExtensionTest {
+
+    @RegisterExtension
+    static QuarkusUnitTest runner = new QuarkusUnitTest()
+            .withApplicationRoot((jar) -> jar
+                    .addClass(MyEntity.class)
+                    .addClass(PersistenceUnitInterceptor.class)
+                    .addClass(AnotherPersistenceUnitInterceptor.class))
+            .withConfigurationResource("application.properties")
+            .assertException(throwable -> assertThat(throwable)
+                    .hasNoSuppressedExceptions()
+                    .rootCause()
+                    .hasMessageContainingAll("Multiple instances of Interceptor were found at ",
+                            "io.quarkus.hibernate.orm.boot.AmbiguousPersistenceUnitExtensionTest.PersistenceUnitInterceptor",
+                            "io.quarkus.hibernate.orm.boot.AmbiguousPersistenceUnitExtensionTest.AnotherPersistenceUnitInterceptor",
+                            "for persistence unit <default>. At most one instance can be assigned to each persistence unit."));
+
+    @PersistenceUnitExtension
+    public static class PersistenceUnitInterceptor implements Interceptor {
+    }
+
+    @PersistenceUnitExtension
+    public static class AnotherPersistenceUnitInterceptor implements Interceptor {
+    }
+
+    @Test
+    public void test() {
+        Assertions.fail("Startup should have failed");
+    }
+}

--- a/extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/boot/AmbiguousPersistenceUnitExtensionTest.java
+++ b/extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/boot/AmbiguousPersistenceUnitExtensionTest.java
@@ -23,10 +23,10 @@ public class AmbiguousPersistenceUnitExtensionTest {
             .assertException(throwable -> assertThat(throwable)
                     .hasNoSuppressedExceptions()
                     .rootCause()
-                    .hasMessageContainingAll("Multiple instances of Interceptor were found at ",
+                    .hasMessageContainingAll("Multiple instances of Interceptor were found for persistence unit <default>.",
+                            "At most one instance can be assigned to each persistence unit. Instances found:",
                             "io.quarkus.hibernate.orm.boot.AmbiguousPersistenceUnitExtensionTest.PersistenceUnitInterceptor",
-                            "io.quarkus.hibernate.orm.boot.AmbiguousPersistenceUnitExtensionTest.AnotherPersistenceUnitInterceptor",
-                            "for persistence unit <default>. At most one instance can be assigned to each persistence unit."));
+                            "io.quarkus.hibernate.orm.boot.AmbiguousPersistenceUnitExtensionTest.AnotherPersistenceUnitInterceptor"));
 
     @PersistenceUnitExtension
     public static class PersistenceUnitInterceptor implements Interceptor {

--- a/extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/PersistenceUnitUtil.java
+++ b/extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/PersistenceUnitUtil.java
@@ -34,9 +34,9 @@ public class PersistenceUnitUtil {
             List<String> ambiguousClassNames = instance.handlesStream().map(h -> h.getBean().getBeanClass().getCanonicalName())
                     .toList();
             throw new IllegalStateException(String.format(Locale.ROOT,
-                    "Multiple instances of %1$s were found at %2$s for persistence unit %3$s. "
-                            + "At most one instance can be assigned to each persistence unit.",
-                    beanType.getSimpleName(), ambiguousClassNames, persistenceUnitName));
+                    "Multiple instances of %1$s were found for persistence unit %2$s. "
+                            + "At most one instance can be assigned to each persistence unit. Instances found: %3$s",
+                    beanType.getSimpleName(), persistenceUnitName, ambiguousClassNames));
         }
         return instance;
     }

--- a/extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/PersistenceUnitUtil.java
+++ b/extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/PersistenceUnitUtil.java
@@ -3,6 +3,7 @@ package io.quarkus.hibernate.orm.runtime;
 import java.lang.annotation.Annotation;
 import java.util.Arrays;
 import java.util.Comparator;
+import java.util.List;
 import java.util.Locale;
 
 import jakarta.enterprise.inject.Default;
@@ -30,10 +31,12 @@ public class PersistenceUnitUtil {
         InjectableInstance<T> instance = extensionInstanceForPersistenceUnit(beanType, persistenceUnitName,
                 additionalQualifiers);
         if (instance.isAmbiguous()) {
+            List<String> ambiguousClassNames = instance.handlesStream().map(h -> h.getBean().getBeanClass().getCanonicalName())
+                    .toList();
             throw new IllegalStateException(String.format(Locale.ROOT,
-                    "Multiple instances of %1$s were found for persistence unit %2$s. "
+                    "Multiple instances of %1$s were found at %2$s for persistence unit %3$s. "
                             + "At most one instance can be assigned to each persistence unit.",
-                    beanType.getSimpleName(), persistenceUnitName));
+                    beanType.getSimpleName(), ambiguousClassNames, persistenceUnitName));
         }
         return instance;
     }

--- a/extensions/hibernate-search-orm-elasticsearch/deployment/src/test/java/io/quarkus/hibernate/search/orm/elasticsearch/test/boot/AmbiguousSearchExtensionTest.java
+++ b/extensions/hibernate-search-orm-elasticsearch/deployment/src/test/java/io/quarkus/hibernate/search/orm/elasticsearch/test/boot/AmbiguousSearchExtensionTest.java
@@ -29,10 +29,11 @@ public class AmbiguousSearchExtensionTest {
             .assertException(throwable -> assertThat(throwable)
                     .hasNoSuppressedExceptions()
                     .rootCause()
-                    .hasMessageContainingAll("Multiple instances of FailureHandler were found at ",
+                    .hasMessageContainingAll(
+                            "Multiple instances of FailureHandler were found for Hibernate Search in persistence unit <default>.",
+                            "At most one instance can be assigned to each persistence unit. Instances found:",
                             "io.quarkus.hibernate.search.orm.elasticsearch.test.boot.AmbiguousSearchExtensionTest.SearchFailureHandler",
-                            "io.quarkus.hibernate.search.orm.elasticsearch.test.boot.AmbiguousSearchExtensionTest.AnotherSearchFailureHandler",
-                            "for Hibernate Search in persistence unit <default>. At most one instance can be assigned to each persistence unit."));
+                            "io.quarkus.hibernate.search.orm.elasticsearch.test.boot.AmbiguousSearchExtensionTest.AnotherSearchFailureHandler"));
 
     @Entity
     @Indexed

--- a/extensions/hibernate-search-orm-elasticsearch/deployment/src/test/java/io/quarkus/hibernate/search/orm/elasticsearch/test/boot/AmbiguousSearchExtensionTest.java
+++ b/extensions/hibernate-search-orm-elasticsearch/deployment/src/test/java/io/quarkus/hibernate/search/orm/elasticsearch/test/boot/AmbiguousSearchExtensionTest.java
@@ -1,0 +1,72 @@
+package io.quarkus.hibernate.search.orm.elasticsearch.test.boot;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.Id;
+
+import org.hibernate.search.engine.reporting.EntityIndexingFailureContext;
+import org.hibernate.search.engine.reporting.FailureContext;
+import org.hibernate.search.engine.reporting.FailureHandler;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.Indexed;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.hibernate.search.orm.elasticsearch.SearchExtension;
+import io.quarkus.test.QuarkusUnitTest;
+
+public class AmbiguousSearchExtensionTest {
+
+    @RegisterExtension
+    static QuarkusUnitTest runner = new QuarkusUnitTest()
+            .withApplicationRoot((jar) -> jar
+                    .addClass(MyEntity.class)
+                    .addClass(SearchFailureHandler.class)
+                    .addClass(AnotherSearchFailureHandler.class)
+                    .addAsResource("application.properties"))
+            .assertException(throwable -> assertThat(throwable)
+                    .hasNoSuppressedExceptions()
+                    .rootCause()
+                    .hasMessageContainingAll("Multiple instances of FailureHandler were found at ",
+                            "io.quarkus.hibernate.search.orm.elasticsearch.test.boot.AmbiguousSearchExtensionTest.SearchFailureHandler",
+                            "io.quarkus.hibernate.search.orm.elasticsearch.test.boot.AmbiguousSearchExtensionTest.AnotherSearchFailureHandler",
+                            "for Hibernate Search in persistence unit <default>. At most one instance can be assigned to each persistence unit."));
+
+    @Entity
+    @Indexed
+    static class MyEntity {
+
+        @Id
+        @GeneratedValue
+        private Long id;
+    }
+
+    @SearchExtension
+    public static class SearchFailureHandler implements FailureHandler {
+        @Override
+        public void handle(FailureContext failureContext) {
+        }
+
+        @Override
+        public void handle(EntityIndexingFailureContext entityIndexingFailureContext) {
+        }
+    }
+
+    @SearchExtension
+    public static class AnotherSearchFailureHandler implements FailureHandler {
+        @Override
+        public void handle(FailureContext failureContext) {
+        }
+
+        @Override
+        public void handle(EntityIndexingFailureContext entityIndexingFailureContext) {
+        }
+    }
+
+    @Test
+    public void test() {
+        Assertions.fail("Startup should have failed");
+    }
+}

--- a/extensions/hibernate-search-orm-elasticsearch/runtime/src/main/java/io/quarkus/hibernate/search/orm/elasticsearch/runtime/bean/HibernateSearchBeanUtil.java
+++ b/extensions/hibernate-search-orm-elasticsearch/runtime/src/main/java/io/quarkus/hibernate/search/orm/elasticsearch/runtime/bean/HibernateSearchBeanUtil.java
@@ -32,19 +32,19 @@ public final class HibernateSearchBeanUtil {
                     .toList();
             if (indexName != null) {
                 throw new IllegalStateException(String.format(Locale.ROOT,
-                        "Multiple instances of %1$s were found at %2$s for Hibernate Search index %3$s in persistence unit %4$s."
-                                + " At most one instance can be assigned to each index.",
-                        beanType.getSimpleName(), ambiguousClassNames, indexName, persistenceUnitName));
+                        "Multiple instances of %1$s were found for Hibernate Search index %2$s in persistence unit %3$s."
+                                + " At most one instance can be assigned to each index. Instances found: %4$s",
+                        beanType.getSimpleName(), indexName, persistenceUnitName, ambiguousClassNames));
             } else if (backendName != null) {
                 throw new IllegalStateException(String.format(Locale.ROOT,
-                        "Multiple instances of %1$s were found at %2$s for Hibernate Search backend %3$s in persistence unit %4$s."
-                                + " At most one instance can be assigned to each backend.",
-                        beanType.getSimpleName(), ambiguousClassNames, backendName, persistenceUnitName));
+                        "Multiple instances of %1$s were found for Hibernate Search backend %2$s in persistence unit %3$s."
+                                + " At most one instance can be assigned to each backend. Instances found: %4$s",
+                        beanType.getSimpleName(), backendName, persistenceUnitName, ambiguousClassNames));
             } else {
                 throw new IllegalStateException(String.format(Locale.ROOT,
-                        "Multiple instances of %1$s were found at %2$s for Hibernate Search in persistence unit %3$s."
-                                + " At most one instance can be assigned to each persistence unit.",
-                        beanType.getSimpleName(), ambiguousClassNames, persistenceUnitName));
+                        "Multiple instances of %1$s were found for Hibernate Search in persistence unit %2$s."
+                                + " At most one instance can be assigned to each persistence unit. Instances found: %3$s",
+                        beanType.getSimpleName(), persistenceUnitName, ambiguousClassNames));
             }
         }
         return instance.isResolvable() ? Optional.of(new ArcBeanReference<>(instance.getHandle().getBean())) : Optional.empty();

--- a/extensions/hibernate-search-orm-elasticsearch/runtime/src/main/java/io/quarkus/hibernate/search/orm/elasticsearch/runtime/bean/HibernateSearchBeanUtil.java
+++ b/extensions/hibernate-search-orm-elasticsearch/runtime/src/main/java/io/quarkus/hibernate/search/orm/elasticsearch/runtime/bean/HibernateSearchBeanUtil.java
@@ -28,21 +28,23 @@ public final class HibernateSearchBeanUtil {
             String persistenceUnitName, String backendName, String indexName) {
         InjectableInstance<T> instance = extensionInstanceFor(beanType, persistenceUnitName, backendName, indexName);
         if (instance.isAmbiguous()) {
+            List<String> ambiguousClassNames = instance.handlesStream().map(h -> h.getBean().getBeanClass().getCanonicalName())
+                    .toList();
             if (indexName != null) {
                 throw new IllegalStateException(String.format(Locale.ROOT,
-                        "Multiple instances of %1$s were found for Hibernate Search index %3$s in persistence unit %2$s."
+                        "Multiple instances of %1$s were found at %2$s for Hibernate Search index %3$s in persistence unit %4$s."
                                 + " At most one instance can be assigned to each index.",
-                        beanType.getSimpleName(), persistenceUnitName, indexName));
+                        beanType.getSimpleName(), ambiguousClassNames, indexName, persistenceUnitName));
             } else if (backendName != null) {
                 throw new IllegalStateException(String.format(Locale.ROOT,
-                        "Multiple instances of %1$s were found for Hibernate Search backend %3$s in persistence unit %2$s."
+                        "Multiple instances of %1$s were found at %2$s for Hibernate Search backend %3$s in persistence unit %4$s."
                                 + " At most one instance can be assigned to each backend.",
-                        beanType.getSimpleName(), persistenceUnitName, backendName));
+                        beanType.getSimpleName(), ambiguousClassNames, backendName, persistenceUnitName));
             } else {
                 throw new IllegalStateException(String.format(Locale.ROOT,
-                        "Multiple instances of %1$s were found for Hibernate Search in persistence unit %2$s."
+                        "Multiple instances of %1$s were found at %2$s for Hibernate Search in persistence unit %3$s."
                                 + " At most one instance can be assigned to each persistence unit.",
-                        beanType.getSimpleName(), persistenceUnitName));
+                        beanType.getSimpleName(), ambiguousClassNames, persistenceUnitName));
             }
         }
         return instance.isResolvable() ? Optional.of(new ArcBeanReference<>(instance.getHandle().getBean())) : Optional.empty();

--- a/extensions/hibernate-search-standalone-elasticsearch/deployment/src/test/java/io/quarkus/hibernate/search/standalone/elasticsearch/test/boot/AmbiguousSearchExtensionTest.java
+++ b/extensions/hibernate-search-standalone-elasticsearch/deployment/src/test/java/io/quarkus/hibernate/search/standalone/elasticsearch/test/boot/AmbiguousSearchExtensionTest.java
@@ -1,0 +1,68 @@
+package io.quarkus.hibernate.search.standalone.elasticsearch.test.boot;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.hibernate.search.engine.reporting.EntityIndexingFailureContext;
+import org.hibernate.search.engine.reporting.FailureContext;
+import org.hibernate.search.engine.reporting.FailureHandler;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.DocumentId;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.Indexed;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.SearchEntity;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.hibernate.search.standalone.elasticsearch.SearchExtension;
+import io.quarkus.test.QuarkusUnitTest;
+
+public class AmbiguousSearchExtensionTest {
+    @RegisterExtension
+    static QuarkusUnitTest runner = new QuarkusUnitTest()
+            .withApplicationRoot((jar) -> jar
+                    .addClass(MyEntity.class)
+                    .addClass(SearchFailureHandler.class)
+                    .addClass(AnotherSearchFailureHandler.class)
+                    .addAsResource("application.properties"))
+            .assertException(throwable -> assertThat(throwable)
+                    .hasNoSuppressedExceptions()
+                    .rootCause()
+                    .hasMessageContainingAll("Multiple instances of FailureHandler were found at ",
+                            "io.quarkus.hibernate.search.standalone.elasticsearch.test.boot.AmbiguousSearchExtensionTest.SearchFailureHandler",
+                            "io.quarkus.hibernate.search.standalone.elasticsearch.test.boot.AmbiguousSearchExtensionTest.AnotherSearchFailureHandler",
+                            "for Hibernate Search Standalone. At most one instance can be assigned."));
+
+    @SearchEntity
+    @Indexed
+    public static class MyEntity {
+        @DocumentId
+        private Long id;
+    }
+
+    @SearchExtension
+    public static class SearchFailureHandler implements FailureHandler {
+        @Override
+        public void handle(FailureContext failureContext) {
+        }
+
+        @Override
+        public void handle(EntityIndexingFailureContext entityIndexingFailureContext) {
+        }
+    }
+
+    @SearchExtension
+    public static class AnotherSearchFailureHandler implements FailureHandler {
+        @Override
+        public void handle(FailureContext failureContext) {
+        }
+
+        @Override
+        public void handle(EntityIndexingFailureContext entityIndexingFailureContext) {
+        }
+    }
+
+    @Test
+    public void test() {
+        Assertions.fail("Startup should have failed");
+    }
+
+}

--- a/extensions/hibernate-search-standalone-elasticsearch/deployment/src/test/java/io/quarkus/hibernate/search/standalone/elasticsearch/test/boot/AmbiguousSearchExtensionTest.java
+++ b/extensions/hibernate-search-standalone-elasticsearch/deployment/src/test/java/io/quarkus/hibernate/search/standalone/elasticsearch/test/boot/AmbiguousSearchExtensionTest.java
@@ -26,10 +26,11 @@ public class AmbiguousSearchExtensionTest {
             .assertException(throwable -> assertThat(throwable)
                     .hasNoSuppressedExceptions()
                     .rootCause()
-                    .hasMessageContainingAll("Multiple instances of FailureHandler were found at ",
+                    .hasMessageContainingAll(
+                            "Multiple instances of FailureHandler were found for Hibernate Search Standalone.",
+                            "At most one instance can be assigned. Instances found:",
                             "io.quarkus.hibernate.search.standalone.elasticsearch.test.boot.AmbiguousSearchExtensionTest.SearchFailureHandler",
-                            "io.quarkus.hibernate.search.standalone.elasticsearch.test.boot.AmbiguousSearchExtensionTest.AnotherSearchFailureHandler",
-                            "for Hibernate Search Standalone. At most one instance can be assigned."));
+                            "io.quarkus.hibernate.search.standalone.elasticsearch.test.boot.AmbiguousSearchExtensionTest.AnotherSearchFailureHandler"));
 
     @SearchEntity
     @Indexed

--- a/extensions/hibernate-search-standalone-elasticsearch/runtime/src/main/java/io/quarkus/hibernate/search/standalone/elasticsearch/runtime/bean/HibernateSearchBeanUtil.java
+++ b/extensions/hibernate-search-standalone-elasticsearch/runtime/src/main/java/io/quarkus/hibernate/search/standalone/elasticsearch/runtime/bean/HibernateSearchBeanUtil.java
@@ -28,21 +28,23 @@ public final class HibernateSearchBeanUtil {
             String backendName, String indexName) {
         InjectableInstance<T> instance = extensionInstanceFor(beanType, backendName, indexName);
         if (instance.isAmbiguous()) {
+            List<String> ambiguousClassNames = instance.handlesStream().map(h -> h.getBean().getBeanClass().getCanonicalName())
+                    .toList();
             if (indexName != null) {
                 throw new IllegalStateException(String.format(Locale.ROOT,
-                        "Multiple instances of %1$s were found for Hibernate Search Standalone index %2$s."
+                        "Multiple instances of %1$s were found at %2$s for Hibernate Search Standalone index %3$s."
                                 + " At most one instance can be assigned to each index.",
-                        beanType.getSimpleName(), indexName));
+                        beanType.getSimpleName(), ambiguousClassNames, indexName));
             } else if (backendName != null) {
                 throw new IllegalStateException(String.format(Locale.ROOT,
-                        "Multiple instances of %1$s were found for Hibernate Search Standalone backend %2$s."
+                        "Multiple instances of %1$s were found at %2$s for Hibernate Search Standalone backend %3$s."
                                 + " At most one instance can be assigned to each backend.",
-                        beanType.getSimpleName(), backendName));
+                        beanType.getSimpleName(), ambiguousClassNames, backendName));
             } else {
                 throw new IllegalStateException(String.format(Locale.ROOT,
-                        "Multiple instances of %1$s were found for Hibernate Search Standalone."
+                        "Multiple instances of %1$s were found at %2$s for Hibernate Search Standalone."
                                 + " At most one instance can be assigned.",
-                        beanType.getSimpleName()));
+                        beanType.getSimpleName(), ambiguousClassNames));
             }
         }
         return instance.isResolvable() ? Optional.of(new ArcBeanReference<>(instance.getHandle().getBean())) : Optional.empty();

--- a/extensions/hibernate-search-standalone-elasticsearch/runtime/src/main/java/io/quarkus/hibernate/search/standalone/elasticsearch/runtime/bean/HibernateSearchBeanUtil.java
+++ b/extensions/hibernate-search-standalone-elasticsearch/runtime/src/main/java/io/quarkus/hibernate/search/standalone/elasticsearch/runtime/bean/HibernateSearchBeanUtil.java
@@ -32,18 +32,18 @@ public final class HibernateSearchBeanUtil {
                     .toList();
             if (indexName != null) {
                 throw new IllegalStateException(String.format(Locale.ROOT,
-                        "Multiple instances of %1$s were found at %2$s for Hibernate Search Standalone index %3$s."
-                                + " At most one instance can be assigned to each index.",
-                        beanType.getSimpleName(), ambiguousClassNames, indexName));
+                        "Multiple instances of %1$s were found for Hibernate Search Standalone index %2$s."
+                                + " At most one instance can be assigned to each index. Instances found: %3$s",
+                        beanType.getSimpleName(), indexName, ambiguousClassNames));
             } else if (backendName != null) {
                 throw new IllegalStateException(String.format(Locale.ROOT,
-                        "Multiple instances of %1$s were found at %2$s for Hibernate Search Standalone backend %3$s."
-                                + " At most one instance can be assigned to each backend.",
-                        beanType.getSimpleName(), ambiguousClassNames, backendName));
+                        "Multiple instances of %1$s were found for Hibernate Search Standalone backend %2$s."
+                                + " At most one instance can be assigned to each backend. Instances found: %3$s",
+                        beanType.getSimpleName(), backendName, ambiguousClassNames));
             } else {
                 throw new IllegalStateException(String.format(Locale.ROOT,
-                        "Multiple instances of %1$s were found at %2$s for Hibernate Search Standalone."
-                                + " At most one instance can be assigned.",
+                        "Multiple instances of %1$s were found for Hibernate Search Standalone."
+                                + " At most one instance can be assigned. Instances found: %2$s",
                         beanType.getSimpleName(), ambiguousClassNames));
             }
         }

